hsmetrics.wdl

```diff
@@ -0,0 +1,160 @@
+version 1.0
+
+workflow hsMetrics {
+input {
+    File    inputBam
+    File    baitBed
+    File    targetBed
+    String  refFasta
+    String? outputPrefix = ""
+    String? stringencyFilter = "LENIENT"
+    Float?  minPct      = 0.5
+    Int?    coverageCap = 500
+}
+
+call makeRefDictionary { input: refFasta = refFasta }
+call bedToIntervals as bedToTargetIntervals { input: inputBed = targetBed, refDict = makeRefDictionary.refDict }
+call bedToIntervals as bedToBaitIntervals { input: inputBed = baitBed, refDict = makeRefDictionary.refDict }
+
+call collectHSmetrics{ input: inputBam = inputBam, baitIntervals = bedToBaitIntervals.outputIntervals, targetIntervals = bedToTargetIntervals.outputIntervals, refFasta = refFasta, filter = stringencyFilter, coverageCap = coverageCap, outputPrefix = outputPrefix }
+call collectInsertMetrics{ input: inputBam = inputBam, minPct = minPct, outputPrefix = outputPrefix }
+
+output {
+  File outputHSMetrics  = collectHSmetrics.outputHSMetrics
+  File outputINSMetrics = collectInsertMetrics.outputINSMetrics
+  File outputINSPDF     = collectInsertMetrics.outputINSPDF
+}
+
+}
+
+# ==========================================
+#  TASK 1 of 4: make reference dictionary
+# ==========================================
+task makeRefDictionary {
+input {
+        String refFasta
+        Int?   jobMemory  = 18
+        Int?   javaMemory = 10
+        String? modules   = "java/8 picard/2.19.2 hg19/p13"
+}
+
+command <<<
+ java -Xmx~{javaMemory}G -jar $PICARD_ROOT/picard.jar CreateSequenceDictionary \
+                              REFERENCE=~{refFasta} \
+                              OUTPUT="~{basename(refFasta, '.fa')}.dict" 
+>>>
+
+runtime {
+  memory:  "~{jobMemory} GB"
+  modules: "~{modules}"
+}
+
+output {
+  File refDict = "~{basename(refFasta, '.fa')}.dict"
+}
+}
+
+# ==========================================
+#  TASK 2 of 4: convert bed to intervals
+# ==========================================
+task bedToIntervals {
```

Comment:
This data should be moved to modulator

Reply:
this is a very light task, and if we want to have these in modulator we will probably want to have all .bed files in modulator as well. Maybe Version 1.1?

Reply:
Sounds good.